plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryWithDifferentProjectIdConnectorTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.bigquery;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.QueryRunner;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestBigQueryWithDifferentProjectIdConnectorTest
```

Comment:
How does this test verify that the objects are getting created in correct project?
i.e. would this test fail without your fix?

Reply:
Yes, a handful of tests fail without the fixes -- `testCreateSchema` and the native query tests as well.

We could add an explicit test for clarity if you'd like.

Reply:
I think an explicit one + converting the new one to extend `BaseConnectorSmokeTest` instead should be sufficient. Note that for configuration changes which don't affect "execution" the smoke test generally provides good enough coverage.

Reply:
Sounds good. I'll revert back so `TestBigQueryConnectorTest` stays as it is today and create a smoke test with a different project ID. `testCreateSchema` fails in the smoke test w/o the fixes but I'll add an explicit one and add a test case for the native table function.